frontend/src/views/instructor/ddahs/import.tsx

```diff
@@ -0,0 +1,319 @@
+}
+
+function dutiesDiffList(changes: DutyDiff[]) {
+    return (
+        <ul>
+            {changes.map(({ oldDuty, newDuty, status }) => {
+                switch (status) {
+                    case DiffType.Unchanged:
+                        return (
+                            <li className="duty-unchanged">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Updated:
+                        return (
+                            <li className="duty-updated">
+                                {oldDuty!.description} - {oldDuty!.hours}h{" "}
+                                <FaArrowRight />
+                                {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Created:
+                        return (
+                            <li className="duty-created">
+                                {newDuty!.description} - {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Deleted:
+                        return (
+                            <li className="duty-deleted">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    default:
+                        return null;
+                }
+            })}
+        </ul>
+    );
+}
+
+function DdahsList({ ddahs }: { ddahs: Omit<Ddah, "id">[] }) {
+    return (
+        <ul>
+            {ddahs.map(({ assignment: { applicant }, duties }, i) => (
+                <li>
+                    DDAH for {applicant.first_name} {applicant.last_name} (
+                    {applicant.utorid})
+                    <ul className="instructor-ddah-import-new-ddah-duties-list">
+                        {duties.map((duty) => (
+                            <li key={`ddah-new-${i}`}>
+                                {duty.description} - {duty.hours}h
+                            </li>
+                        ))}
+                    </ul>
+                </li>
+            ))}
+        </ul>
+    );
+}
+
+function DdahsDiffList({ ddahUpdates }: { ddahUpdates: DdahUpdate[] }) {
+    return (
+        <ul className="instructor-ddah-import-duties-diff-list">
+            {ddahUpdates.map(({ oldDdah, updatedDdah }, i) => {
+                const dutiesDiff = getDutiesDiff(
+                    oldDdah.duties,
+                    updatedDdah.duties
+                );
+                const {
+                    assignment: {
+                        applicant: { first_name, last_name, utorid },
+                    },
+                } = updatedDdah;
+                return (
+                    <li key={`ddah-diff-${i}`}>
+                        DDAH for {first_name} {last_name} ({utorid})
+                        {dutiesDiffList(dutiesDiff)}
```

Comment:
Why aren't you using a JSX component? You're bypassing any optimizations that React chooses to make when you do it this way.